crates/adapters/src/integrated/delta_table/input.rs

```diff
@@ -556,6 +617,118 @@ impl DeltaTableInputEndpointInner {
         Ok(())
     }
 
+    /// Validate the cdc_order_by expression.
+    fn validate_cdc_order_by(&self) -> Result<(), ControllerError> {
+        if let Some(order_by) = &self.config.cdc_order_by {
+            validate_sql_expression(order_by).map_err(|e| {
+                ControllerError::invalid_transport_configuration(
+                    &self.endpoint_name,
+                    &format!("error parsing 'cdc_order_by' expression '{order_by}': {e}"),
+                )
+            })?;
+        }
+
+        Ok(())
+    }
+
+    /// Convert the delete filter SQL expression into a Datafusion PhysicalExpr.
+    ///
+    /// To do this, we generate a query plan for the `select * from snapshot where <delete_filter>`
+    /// query and extract the logical expression from the plan and then convert it to
+    /// a physical expression.
+    //
+    // FIXME: I wonder if there's a simpler way that doesn't require registering the `snapshot`
+    // table just so we can compile this query.
+    async fn extract_delete_filter_expr(
+        &self,
+    ) -> Result<Option<Arc<dyn PhysicalExpr>>, ControllerError> {
+        let Some(delete_filter) = &self.config.cdc_delete_filter else {
+            return Ok(None);
+        };
+
+        let query = format!("SELECT * FROM snapshot WHERE {}", delete_filter);
+
+        let logical_plan = self
+            .datafusion
+            .sql(&query)
+            .await
+            .map_err(|e| {
+                ControllerError::invalid_transport_configuration(
+                    &self.endpoint_name,
+                    &format!("invalid delete filter '{delete_filter}': the 'cdc_delete_filter' expression must be a valid SQL expression that can be used in a 'SELECT * FROM snapshot WHERE <cdc_delete_filter>' query, but the following error was encountered when compiling '{query}': {e}"),
+                )
+            })?
+            .logical_plan()
+            .clone();
+
+        let filter_expr = if let LogicalPlan::Projection(filter_plan) = &logical_plan {
+            if let LogicalPlan::Filter(filter) = filter_plan.input.as_ref() {
+                filter.predicate.clone()
```

Comment:
I am doing this kind of stuff too in Calcite, but you have to be careful, because for filters like 'true' the optimizer may remove the filter altogether.